common/types.go

```diff
@@ -712,17 +712,20 @@ func ConstructError(statusCode int, b []byte) error {
 	var err struct {
 		Error *Error
 	}
-	if e := json.Unmarshal(b, &err); e != nil || err.Error == nil{
+	if e := json.Unmarshal(b, &err); e != nil || err.Error == nil {
 		// return normal error
 		return fmt.Errorf("%d: %s", statusCode, string(b))
```

Comment:
I wonder if it might be better to always return the Error object. We could handle this unmarshalling error by manually setting err.Error.Message = string(b). That way any consuming client wouldn't need to have conditional checking on what was returned. It can always be assumed to be a Redfish Error object, even when another error has occurred. Then they can always access the error message the same way.